torchtune/data/_converters.py

```diff
@@ -62,6 +66,9 @@ def get_sharegpt_messages(
     return messages
 
 
+@deprecated(
+    msg="Please use `torchtune.data.JSONToMessages` with `torchtune.datasets.SFTDataset` instead."
+)
```

Comment:
Maybe slight overkill (and also the usefulness of this depends on how much people actually use a given API, which we don't have clear insights into), but for any migrations that are nontrivial we could consider updating the docstring with a quick example of how to make the change in user code.

Reply:
yeah let's do that. can add a small note to all the docstrings of deprecated classes

Reply:
I'm going to leave this out actually, a lot of concepts (like prompt template) have moved out of the dataset class completely, and it might be more confusing to try to show hard equivalents